### DISCUSSION TITLE
Tags for Headings instead of Entire Pages

### DISCUSSION THREAD

**Discussion by aFineBoi** [zim-desktop-wiki/zim-desktop-wiki#2150] [Ideas]
For long pages, tags for a whole page are of no good and tags for content (like headings and anchors) within a page are needed instead.

Currently the solution is to split the page into many, many smaller files when multiple tags are needed for individual pieces of content, but this is very counterintuitive (believe me, you do not want to split hundreds of small records in a single page into hundreds of .txt files).

**Comment by downtoearthman** [comment]
I believe you can do what you are requesting at least within the same notebook. I did not find this solution anywhere in the manual, but just found it by experimenting. This is how you do it:
Once you have created a subheading with an h1, h2, h3, h4 etc. then triple click on it or drag over it holding the left mouse button down to select it:
I also found you can just right click on it without selecting it:
So right click on it and select the "Copy link to this location" option.
Then go to whatever page you want, within the same notebook, and right click and paste the link wherever you want it.
For me this works as long as I am in the same notebook.
I am using a Linux computer, so if you are on windows or mac, then you might have to triple click or drag your mouse to select and make it work.

I want to be able to link to headings in a different notebook and it doesn't work.
For example I copied the zim manual from usr/share/zim/manual into my notebooks folder:
Then in the notebook.zim file I set the interwiki setting to interwiki=Help
There is a page that has an h4 like this: GUI_style_config_file
At first spaces occupied the place where the underscores are, but I know you can't have spaces in a link you type.
So this is what I typed: Help?Help:Config_Files#GUI_style_config_file
I did this from another notebook and it did not work.
I hope that someone can figure out a way to link to subheading like this:
I know I can link to pages from another notebook. 
For example:
When I link to a page like this:
Help?Help:Config_Files
It works, but then I have to Copy the link to the heading location and then paste that link to the top of that page like this:
GUI_style_config_file
and then click on that link to go to that heading.
I think it would be nice to be able to link to a heading from another notebook with something like this:
Help?Help:Config_Files#GUI_style_config_file
I hope the zim programmers will see this request or if someone can show me how this is done.

**Reply by introt** [reply]
> I want to be able to link to headings in a different notebook and it doesn't work.

This should be fixed in 0.75, please do give it a try

**Comment by introt** [comment]
For that kind of data you might find another tool like [TiddlyWiki](https://tiddlywiki.com/) preferable. Its unique approach allows you to split your long pages into small _tiddlers_, which you can _transclude_ into another tiddler either manually or automatically (based on their tags etc) to produce "whole pages" from hundreds of "small records".

**Reply by downtoearthman** [reply]
Wow TiddlyWiki looks like the very thing I've been looking for... for years. This does not fit exactly what I was trying to achieve in my question, but it is what is behind the root idea of what inspired my question or comment above. The truth is that I've been wanting to figure out a system that can interweave my ideas, intentions, goals, knowledge base, understanding and all things thought process in such a way that can help me to put my ideas together. It appears that TiddlyWiki may not be perfect, but it's a far cry better than anything I've put together so far. I have about 50 Zim Desktop Wiki notebooks which I'm trying to put all my ideas, projects and whatever I'm involved with together. 

I did a search and found Soren Bjornstad's book Titled: Grok I can see there is already something out there that is light years ahead of me. I will rebuild it in Node.js / Vue.js frontend to customize it to serve my own purposes, but I know that learning this concept and the idea behind Soren's book will help me to put together the knowledge base I want to achieve. I was already looking at the shortcomings of Zim Desktop Wiki and planning to create my own note taking system with Node/Vue, but now I can see that TiddlyWiki will save me probly years of trial and error to get to the kind of note system that can help the human mind put things together.

I usually never want to repay someone when I get info like this for free on the internet, but if there is any way that I can repay you for this introduction to TiddlyWiki I would love to. If there is any way I can help you, let me know. Of course you can contact me through my user here on github. Thanks so much introt

**Reply by introt** [reply]
You're most welcome @downtoearthman!

The best way to repay me is to pay it forward - help others whenever you have the chance. Open-sourcing your work is one way to share the knowledge you've gathered along the way. I personally struggle with this due to perfectionism, yet encourage others to do so regardless of their tools' completeness: even half-baked solutions might prove useful to others working on the same problems.

Happy Hacking!

**Reply by downtoearthman** [reply]
Thanks for your reply... as a node.js and JavaScript programmer I have been looking for a good Open-source project to contribute to. I may even contribute to the Tiddlywiki documentation as I've been learning it and have seen some ways I can help to improve it..

Thanks again and I will pay it forward.
